src/movement/test/test_thruster_sabertooth.cpp

```diff
@@ -0,0 +1,120 @@
+#include <gtest/gtest.h>
+#include "ros/ros.h"
+#include "robosub/thruster.h"
+#include "utility/serial_testbench.hpp"
+#include "utility/serial.hpp"
+
+typedef struct thruster_info
+{
+  std::string name;
+  uint8_t address;
+  uint8_t port;      // 0 or 1
+}Thruster_info;
+
+Thruster_info *mThruster_info;
+Serial mSerial;
+
+TEST(SerialSubscriber, basicTest)
+{
+    //Variables for testing use.
+    uint8_t address = 0, cmd_byte = 0, speed = 0, checksum = 0;
+    int count = 0;
+    //allocate an array of bytes that will be used to receive serial data from
+    //the serial port. I use the arbitrarily large size of
+    uint8_t serial_data[256] = {0};
```

Comment:
Shouldn't we know exactly how many bytes we should be receiving? If the number of bytes is greater or less than this, the packet is invalid anyways. Is there a use to having an arbitrarily large array?